packages/stacked_generator/lib/src/generators/router/generator/arguments_class/arguments_class_builder.dart

```diff
@@ -10,16 +10,31 @@ class ArgumentsClassBuilder {
   const ArgumentsClassBuilder({required this.routes});
 
   Iterable<Class> buildViewsArguments() {
-    final routesWithParameters =
-        routes.where((route) => notQueryNorPath(route.parameters).isNotEmpty);
+    final routesWithParameters = routes.where(
+      (route) => notQueryNorPath(route.parameters).isNotEmpty,
+    );
 
     return routesWithParameters.map((route) {
       final argumentsBuilderHelper = ArgumentClassBuilderHelper(route);
+
+      final parametersAsString =
+          argumentsBuilderHelper.convertParametersToStrings.toString();
+
       return Class(
         (b) => b
           ..name = argumentsBuilderHelper.argumentClassName
           ..fields.addAll(argumentsBuilderHelper.convertParametersToClassFields)
-          ..constructors.add(argumentsBuilderHelper.argumentConstructer),
+          ..constructors.add(argumentsBuilderHelper.argumentConstructer)
+          ..methods.add(
```

Comment:
Add another function called toMap() that allows you to return the arguments as a map.